.github/workflows/rust.yml

```diff
@@ -1,11 +1,40 @@
 name: Rust
 on:
   pull_request:
+    paths-ignore:
+      - '**.md'
     branches: [develop]
 jobs:
   build:
     runs-on: ubuntu-latest
     steps:
-    - uses: actions/checkout@v1
+    - uses: actions/checkout@v2
+    - uses: actions/cache@v2
+      with:
+        path: |
+          ~/.cargo/registry
+          ~/.cargo/git
+          target
+        # you can edit the .github/cache_bust file if you need to clear the cache
+        key: ${{ hashFiles('.github/cache_bust') }}-${{ runner.os }}-build-${{ hashFiles('**/Cargo.lock') }}
+        restore-keys: |
+          ${{ hashFiles('.github/cache_bust') }}-${{ runner.os }}-build-
     - run: rustup update stable
-    - run: make ci
+    - run: make build
+
+  integ:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - uses: actions/cache@v2
+        with:
+          path: |
+            ~/.cargo/registry
+            ~/.cargo/git
+            target
+          # you can edit the .github/cache_bust file if you need to clear the cache
+          key: ${{ hashFiles('.github/cache_bust') }}-${{ runner.os }}-integ-${{ hashFiles('**/Cargo.lock') }}
+          restore-keys: |
+            ${{ hashFiles('.github/cache_bust') }}-${{ runner.os }}-integ-
+      - run: rustup update stable
```

Comment:
Can we make rustup run once before both of the main jobs?  Seems weird / dangerous to try it in parallel.

Reply:
The jobs are isolated in in separate VMs or containers.